linera-rpc/src/grpc_network.rs

```diff
@@ -356,7 +356,9 @@ where
         let pool = ConnectionPool::default();
 
         while let Some((cross_chain_request, shard_id)) = receiver.next().await {
-            if rand::thread_rng().gen::<f32>() < cross_chain_sender_failure_rate {
+            if cross_chain_sender_failure_rate > 0.0
+                && rand::thread_rng().gen::<f32>() < cross_chain_sender_failure_rate
```

Comment:
We could use [`gen_bool`](https://docs.rs/rand/latest/rand/trait.Rng.html#method.gen_bool) here (same in `simple_network`):

```suggestion
                && rand::thread_rng().gen_bool(cross_chain_sender_failure_rate)
```

Reply:
Nice!

Reply:
From what what I gather, `gen_bool(1.0)` is optimized but not `gen_bool(0.0)`.

Reply:
Incidentally, `gen_bool` uses a `f64` so I'm going to stick with `gen::<f32>`.